From 9b65aa4370f9bf8d7fe671e3d8bbc2ec05783919 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Mon, 9 May 2022 20:40:07 -0500
Subject: [PATCH 14/18] cpufreq: ti-cpufreq: Do not print error for
 EPROBE_DEFER

It is possible for dev_pm_opp_set_supported_hw to return EPROBE_DEFER,
so avoid printing out an error message in this case as this is not a
true error and we will be probing again later.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
---
 drivers/cpufreq/ti-cpufreq.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/cpufreq/ti-cpufreq.c b/drivers/cpufreq/ti-cpufreq.c
index 8f9fdd864391..f4a2d5436fc3 100644
--- a/drivers/cpufreq/ti-cpufreq.c
+++ b/drivers/cpufreq/ti-cpufreq.c
@@ -373,9 +373,10 @@ static int ti_cpufreq_probe(struct platform_device *pdev)
 	ti_opp_table = dev_pm_opp_set_supported_hw(opp_data->cpu_dev,
 						   version, VERSION_COUNT);
 	if (IS_ERR(ti_opp_table)) {
-		dev_err(opp_data->cpu_dev,
-			"Failed to set supported hardware\n");
 		ret = PTR_ERR(ti_opp_table);
+		if (ret != -EPROBE_DEFER)
+			dev_err(opp_data->cpu_dev,
+				"Failed to set supported hardware %d\n", ret);
 		goto fail_put_node;
 	}
 
-- 
2.30.2

